types/V1.mli

```diff
@@ -48,10 +48,23 @@ open Result
 
     {e Release %%VERSION%% } *)
 
-(** {1 Abstract devices}
+type 'a pp = Format.formatter -> 'a -> unit
```

Comment:
and likewise here, this is `Fmt.t` -- while we are trying to minimise dependencies here, Fmt is standalone and has a number of other useful functions to sensibly compose formatters.  I'm not against this patch either though, just a suggestion.